decisions/DR007-Hierarchical-or-compositional-traits-for-specifications.md

```diff
@@ -0,0 +1,361 @@
+- traits can be shared between specifications easily without needing to hide them as emergent
+  phenomena based on well-known property names
+- common traits can be shared with the OpenAssetIO community without having to exactly match a
+  whole specification, decreasing the risk of inclusion of very well known traits (such as `blob`)
+  in the core API if required.
+
+#### Cons
+
+- no guarantee that the traits aren't contradictory
+- working with traits could be more complex than working with a single parent hierarchy due to
+  multiple combinations of traits being possible
+- duplication of traited specification is possible
+- not yet practically validated
+- changes to specific traits still affects all specifications that incorporate them
+
+Estimated cost: Small|Medium|Large
+
+## Outcome
+
+We will adopt Option 2 and implement it in the C++ specification API.
+
+## Rationale
+
+Using traits changes the meaning of the specification from classification to a purpose-driven,
+exact requirement approach. It increases the shareability and extensibility of specifications
+through the modular traits, and avoids hidden traits being expressed through “magic” groups of
+well-known properties where the hierarchy precluded them being ancestors. Although it could
+potentially add complexity in terms of mapping traits to specifications, the ability to search by
+exactly the aspect of the specification that is required could reduce having to traverse over
+classifications while guaranteeing the behaviour can be applied.
+
+## Appendix
+
+### An implementation sketch of Option 2
+
+```python
+from openassetio import Specification, specifications
+
+"""
+EntitySpecification uses:
+* Publishing - schema identifiers should be persisted in the AMS.
+* Some relationship queries
+* managementPolicy
+* Asset browser UIs - in particular, previously persisted schema identifiers.
+"""
+
+
+class BlobTrait(specifications.EntityTrait):
+    _type = "blob"
+    mimeType = Specification.TypedProperty(str, doc="")
+
+
+class ContainerTrait(specifications.EntityTrait):
+    _type = "container"
+
+
+class ImageTrait(specifications.EntityTrait):
+    _type = "image"
+
+    channelNames = Specification.TypedProperty(list, doc="")
+    colorSpace = Specification.TypedProperty(str, doc="")
+
+
+class RasterTrait(specifications.EntityTrait):
+    _type = "raster"
+
+    aspectRatio = Specification.TypedProperty(list, doc="")
+    resolution = Specification.TypedProperty(list, doc="")  # E.g. [“1920x1080”, “420x420”]
+
+
+class GeometryTrait(specifications.EntityTrait):
+    _type = "geometry"
+
+    # startFrame==endFrame means "not animated"
+    startFrame = Specification.TypedProperty(float, doc="")
+    endFrame = Specification.TypedProperty(float, doc="")
+
+
+class EditorialTrait(specifications.EntityTrait):
+    _type = "editorial"
+
+    startFrame = Specification.TypedProperty(int, doc="")
+    endFrame = Specification.TypedProperty(int, doc="")
```

Comment:
[fix] The properties here seem astray? Would they be on a trait definition for a `FrameSequenceTrait`?

Reply:
Replaced with diagrams in 78c87c0d7038ae6c2f948e0d7ed5bc39023d108f . These diagrams have changed the traits available to have a sequence trait.